media/server/gstplayer/source/GstDecryptor.cpp

```diff
@@ -0,0 +1,298 @@
+ * limitations under the License.
+ */
+
+#include "GstDecryptorElementFactory.h"
+#include "GstDecryptorPrivate.h"
+#include "RialtoServerLogging.h"
+
+#include <gst/base/gstbasetransform.h>
+#include <gst/gst.h>
+
+G_BEGIN_DECLS
+
+#define GST_RIALTO_DECRYPTOR_TYPE (gst_rialto_decryptor_get_type())
+#define GST_RIALTO_DECRYPTOR(obj) (G_TYPE_CHECK_INSTANCE_CAST((obj), GST_RIALTO_DECRYPTOR_TYPE, GstRialtoDecryptor))
+#define GST_RIALTO_DECRYPTOR_CLASS(klass)                                                                              \
+    (G_TYPE_CHECK_CLASS_CAST((klass), GST_RIALTO_DECRYPTOR_TYPE, GstRialtoDecryptorClass))
+
+typedef struct _GstRialtoDecryptor GstRialtoDecryptor;
+typedef struct _GstRialtoDecryptorClass GstRialtoDecryptorClass;
+typedef struct firebolt::rialto::server::GstRialtoDecryptorPrivate GstRialtoDecryptorPrivate;
+
+GType gst_rialto_decryptor_get_type(void); // NOLINT(build/function_format)
+
+struct _GstRialtoDecryptor
+{
+    GstBaseTransform parent;
+    GstRialtoDecryptorPrivate *priv;
+};
+
+struct _GstRialtoDecryptorClass
+{
+    GstBaseTransformClass parentClass;
+};
+
+G_END_DECLS
+
+static GstStaticPadTemplate sinkTemplate =
+    GST_STATIC_PAD_TEMPLATE("sink", GST_PAD_SINK, GST_PAD_ALWAYS, GST_STATIC_CAPS_ANY);
+
+static GstStaticPadTemplate srcTemplate =
+    GST_STATIC_PAD_TEMPLATE("src", GST_PAD_SRC, GST_PAD_ALWAYS, GST_STATIC_CAPS_ANY);
+
+GST_DEBUG_CATEGORY(gst_rialto_decryptor_debug_category);
+#define GST_CAT_DEFAULT gst_rialto_decryptor_debug_category
+
+#define gst_rialto_decryptor_parent_class parent_class
+G_DEFINE_TYPE_WITH_PRIVATE(GstRialtoDecryptor, gst_rialto_decryptor, GST_TYPE_BASE_TRANSFORM);
+
+static void gst_rialto_decryptor_finalize(GObject *);                   // NOLINT(build/function_format)
+static GstCaps *gst_rialto_decryptor_transform_caps(GstBaseTransform *, // NOLINT(build/function_format)
+                                                    GstPadDirection, GstCaps *, GstCaps *);
+static GstFlowReturn gst_rialto_decryptor_transform_ip(GstBaseTransform *base, // NOLINT(build/function_format)
+                                                       GstBuffer *buffer);
+
+static void gst_rialto_decryptor_class_init(GstRialtoDecryptorClass *klass) // NOLINT(build/function_format)
+{
+    GST_DEBUG_CATEGORY_INIT(gst_rialto_decryptor_debug_category, "rialtodecryptor", 0, "Decryptor for Rialto");
+
+    GObjectClass *gobject_class = G_OBJECT_CLASS(klass);
+    gobject_class->finalize = GST_DEBUG_FUNCPTR(gst_rialto_decryptor_finalize);
+
+    GstElementClass *element_class = GST_ELEMENT_CLASS(klass);
+    gst_element_class_add_pad_template(element_class, gst_static_pad_template_get(&sinkTemplate));
+    gst_element_class_add_pad_template(element_class, gst_static_pad_template_get(&srcTemplate));
+    gst_element_class_set_static_metadata(element_class, "Rialto Decryptor", GST_ELEMENT_FACTORY_KLASS_DECRYPTOR,
+                                          "Decryptor for Rialto.", "Sky");
```

Comment:
Pedantic: The last field is supposed to be name & email isn't it? If you don't want to take all the blame you can add some others from the team in there too!

Reply:
You're right, added me and Adam.